src/data/list/basic.lean

```diff
@@ -2093,6 +2093,22 @@ end
 | b []      l₂ := rfl
 | b (a::l₁) l₂ := by simp only [cons_append, foldr_cons, foldr_append b l₁ l₂]
 
+theorem foldl_empty [is_empty β] (f : α → β → α) (a : α) : Π l : list β, foldl f a l = a
+| []     := rfl
+| (b::l) := is_empty_elim b
```

Comment:
Do we have the following anywhere?
```lean
-- TODO[gh-6025]: make this an instance once safe to do so
def unique_of_is_empty [is_empty α] : unique (list α) := sorry
```
(xref #6025 for why we shouldn't make this an instance)

Reply:
I can't find such a result anywhere in mathlib. I'll PR this separately.

Reply:
The nice think about such a lemma is you can then just rewrite `l` to `[]` immediately, and you don't need this lemma.

Reply:
I've made this into PR #12582, and removed this lemma from the PR.